### ISSUE TITLE
Security issue: whole content of app.json in build result for Web

### ISSUE THREAD

**Issue by jer-sen** [expo/expo-cli#3810]
### Summary

When building for web, some useless and potentially sensitive (such as hooks configuration with auth tockens (Sentry, ...), project folder on local computer, …) data from app.json are included in the bundle.

### Environment

Expo CLI 4.10.1 environment info:
System:
OS: Windows 10 10.0.19042
Binaries:
Node: 14.15.5 - C:\Program Files\nodejs\node.EXE
Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
npm: 7.10.0 - C:\Program Files\nodejs\npm.CMD
IDEs:
Android Studio: Version 3.5.0.0 AI-191.8026.42.35.5900203
npmPackages:
expo: ~42.0.1 => 42.0.3
react: 16.13.1 => 16.13.1
react-dom: 16.13.1 => 16.13.1
react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2
react-native-web: ~0.13.12 => 0.13.18
Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

web

### Error output

None

### Reproducible demo or steps to reproduce from a blank project

Easy to test with a new project, build for web, look into the JS chunks for “staticConfigPath” or any configuration added to app.json.

**Comment by EvanBacon** [comment]
I've opened a PR to remove the hooks property from the public manifest, are there any other properties that you are concerned about?

**Comment by jer-sen** [comment]
Yes, at least staticConfigPath projectRoot packageJsonPath (so _internal ?). Could you select same properties than for iOS/Android builds? https://github.com/expo/expo-cli/blob/1375d6eea5dd112331244d6800996c62c983b0da/packages/config/src/Config.ts#L136

**Comment by EvanBacon** [comment]
I don't understand the last part, this particular PR is for webpack platforms (web only for now), we can open another for Metro later.

**Comment by jer-sen** [comment]
There is already some code for iOS/Android builds that take only some properties of `app.json`. It would be a good idea to be DRY and reuse the same code. Don't you think? Also it would be nice to get the same values with `expo-cconstant` from web than from iOS/Android.

As for the link I've copied/pasted, it's a piece of code I've found that seems to do already what is needed for web builds. Can you reuse that?

**Comment by jer-sen** [comment]
@EvanBacon any update on this critical issue?

**Comment by jer-sen** [comment]
@EvanBacon?
